controllers/machine_controller.go

```diff
@@ -534,21 +535,20 @@ func (r *MachineReconciler) drainNode(ctx context.Context, cluster *clusterv1.Cl
 		},
 		Out:    writer{klog.Info},
 		ErrOut: writer{klog.Error},
-		DryRun: false,
 	}
 
 	if noderefutil.IsNodeUnreachable(node) {
 		// When the node is unreachable and some pods are not evicted for as long as this timeout, we ignore them.
 		drainer.SkipWaitForDeleteTimeoutSeconds = 60 * 5 // 5 minutes
 	}
 
-	if err := kubedrain.RunCordonOrUncordon(ctx, drainer, node, true); err != nil {
+	if err := kubedrain.RunCordonOrUncordon(drainer, node, true); err != nil {
 		// Machine will be re-reconciled after a cordon failure.
 		log.Error(err, "Cordon failed")
 		return ctrl.Result{}, errors.Errorf("unable to cordon node %s: %v", node.Name, err)
 	}
 
-	if err := kubedrain.RunNodeDrain(ctx, drainer, node.Name); err != nil {
+	if err := kubedrain.RunNodeDrain(drainer, node.Name); err != nil {
```

Comment:
Is the context not yet upstream?

Reply:
Didn't make the change for the other one :(

Reply:
Let's open another one, might be picked for next release, which is ok

Reply:
the context seems to be included in 1.21

Reply:
The node drain portion was not using the context passed in the `DrainHelper` object. kubernetes/kubernetes#100439 uses the passed context for all the drain methods as well.